x-pack/plugins/stack_connectors/server/connector_types/inference/index.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  SubActionConnectorType,
+  ValidatorType,
+} from '@kbn/actions-plugin/server/sub_action_framework/types';
+import {
+  GenerativeAIForSearchPlaygroundConnectorFeatureId,
+  GenerativeAIForSecurityConnectorFeatureId,
+} from '@kbn/actions-plugin/common';
+import { ValidatorServices } from '@kbn/actions-plugin/server/types';
+import { GenerativeAIForObservabilityConnectorFeatureId } from '@kbn/actions-plugin/common/connector_feature_config';
+import { InferenceTaskType } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+import {
+  INFERENCE_CONNECTOR_TITLE,
+  INFERENCE_CONNECTOR_ID,
+  ServiceProviderKeys,
+  SUB_ACTION,
+} from '../../../common/inference/constants';
+import { ConfigSchema, SecretsSchema } from '../../../common/inference/schema';
+import { Config, Secrets } from '../../../common/inference/types';
+import { InferenceConnector } from './inference';
+import { unflattenObject } from '../lib/unflatten_object';
+
+const deleteInferenceEndpoint = async (
+  inferenceId: string,
+  taskType: InferenceTaskType,
+  logger: Logger,
+  esClient: ElasticsearchClient
+) => {
+  try {
+    await esClient.inference.delete({
+      task_type: taskType,
+      inference_id: inferenceId,
+    });
+    logger.info(
```

Comment:
How often do we expect this to get called?  Typically folks don't create too many connectors, so if this doesn't happen often `info` is fine.  If it is called a lot, it'll be noisy, and an issue will get opened to set it to `debug` level.